samples/opengl/depthMapTriangulation/image_triangulator.h

```diff
@@ -0,0 +1,164 @@
+#pragma once
+
+// for later reference
+//#include <OpenMesh/Tools/Decimater/DecimaterT.hh>
+
+#include "saiga/core/geometry/openMeshWrapper.h"
+#include "saiga/opengl/assets/all.h"
+#include "saiga/vision/VisionTypes.h"  // camera parameters
+
+// for later reference
+//#include <set>
+
+#include "imageProcessor.h"
+
+namespace Saiga
+{
+class Triangulator
+{
+    virtual void triangulate_image(ImageView<float> depthImage, OpenTriangleMesh& mesh_out) = 0;
+};
+
+
+// ------------------- naive triangulation -------------------
+
+class SimpleTriangulator : public Triangulator
+{
+   public:
+    using MyMesh = OpenTriangleMesh;
+
+    struct Settings
+    {
+        // the value used for pixels that contain failed depth measurements or got discarded
+        float broken_values = 0.0f;
+
+        StereoCamera4Base<float> cameraParameters = StereoCamera4Base<float>(
+            5.3887405952849110e+02, 5.3937051275591125e+02, 3.2233507920081263e+02, 2.3691517848391885e+02, 40.0f);
+    };
+
+   public:
+    SimpleTriangulator(Settings settings_in);
+    ~SimpleTriangulator();
+
+    void triangulate_image(ImageView<float> depthImage, MyMesh& mesh_out);
```

Comment:
add `override`.
This should also have the exact same syntax as the abstaract function.
So don't use `myMesh` here